packages/react/src/components/Grid/Grid.js

```diff
@@ -8,30 +8,55 @@
 import { settings } from 'carbon-components';
 import cx from 'classnames';
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useContext } from 'react';
+import { useFeatureFlag } from '../FeatureFlags';
 
 const { prefix } = settings;
 
+const GridContext = React.createContext({ hasGridParent: false });
+
 function Grid({
   as: BaseComponent = 'div',
   condensed = false,
   narrow = false,
   fullWidth = false,
+  subgrid,
```

Comment:
Is there a good use-case for manually specifying something as a grid? Was trying to think through when someone might want to do this this 🤔  Seemed like it doing it automatically would cover most situations.

Reply:
My thought was actually the opposite, ensuring we provide a way to opt-out of a nested <Grid> automatically being a subgrid.

```js
<Grid>
  <Grid>Automatically a subgrid</Grid>
  <Grid subgrid={false}>but I want to opt out</Grid>
</Grid>
```

Reply:
@tay1orjones do you have a situation in mind where that happens? (when someone would want to opt-out) Just trying to think of a use-case

Reply:
@joshblack After thinking it through some more and playing with it in the storybook I don't think there's a good reason to have it as a configurable prop. I'll remove it 👍